homeassistant/components/synology_dsm/camera.py

```diff
@@ -83,29 +123,53 @@ def supported_features(self) -> int:
     @property
     def is_recording(self):
         """Return true if the device is recording."""
-        return self._camera.is_recording
+        return self.coordinator.data[self._camera_id].is_recording
 
     @property
     def motion_detection_enabled(self):
         """Return the camera motion detection status."""
-        return self._camera.is_motion_detection_enabled
+        return self.coordinator.data[self._camera_id].is_motion_detection_enabled
 
     def camera_image(self) -> bytes:
         """Return bytes of camera image."""
+        _LOGGER.debug(
+            "SynoDSMCamera.camera_image(%s)",
+            self.coordinator.data[self._camera_id].name,
+        )
         if not self.available:
             return None
-        return self._api.surveillance_station.get_camera_image(self._camera.id)
+        try:
+            return self._api.surveillance_station.get_camera_image(self._camera_id)
+        except (SynologyDSMAPIErrorException) as err:
+            _LOGGER.debug(
+                "SynoDSMCamera.camera_image(%s) - Exception:%s",
+                self.coordinator.data[self._camera_id].name,
+                err,
+            )
+            return None
 
     async def stream_source(self) -> str:
         """Return the source of the stream."""
+        _LOGGER.debug(
+            "SynoDSMCamera.stream_source(%s)",
+            self.coordinator.data[self._camera_id].name,
+        )
         if not self.available:
             return None
-        return self._camera.live_view.rtsp
+        return self.coordinator.data[self._camera_id].live_view.rtsp
```

Comment:
```
@property camera_data(self):
     """Camera data."""
     return self.coordinator.data[self._camera_id]
```

And then replace `self.coordinator.data[self._camera_id]` with `self.camera_data` everywhere.

Reply:
thanks @bdraco again for that improvement idea 😃